src/main/java/org/folio/circulation/domain/OverduePeriodCalculatorService.java

```diff
@@ -0,0 +1,134 @@
+package org.folio.circulation.domain;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.domain.policy.OverdueFinePolicy;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.joda.time.LocalTime;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ResultBinding.flatMapResult;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_DAY;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_HOUR;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_WEEK;
+import static org.joda.time.Minutes.minutesBetween;
+
+public class OverduePeriodCalculatorService {
+  private static final int ZERO_MINUTES = 0;
+  private static final int MINUTES_PER_MONTH = 44640;
+
+  private final CalendarRepository calendarRepository;
+
+  public OverduePeriodCalculatorService(CalendarRepository calendarRepository) {
+    this.calendarRepository = calendarRepository;
+  }
+
+  public static OverduePeriodCalculatorService using(Clients clients) {
+    return new OverduePeriodCalculatorService(new CalendarRepository(clients));
+  }
+
+  public CompletableFuture<Result<Integer>> getMinutes(Loan loan, DateTime systemTime) {
+    if (preconditionsAreNotMet(loan, systemTime)) {
+      return completedFuture(succeeded(ZERO_MINUTES));
+    }
+    return getOverdueMinutes(loan, systemTime)
```

Comment:
Personally, I think this is more readable if there is clear whitespace between the guard clause and the typical return statement

Reply:
done